libraries/AP_OSD/AP_OSD_Backend.cpp

```diff
@@ -24,32 +24,63 @@ constexpr uint8_t AP_OSD_Backend::symbols[AP_OSD_NUM_SYMBOLS];
 #define SYM_DIG_OFS_1 0x90
 #define SYM_DIG_OFS_2 0xA0
 
+
+uint8_t AP_OSD_Backend::convert_to_decimal_packed_characters(char* buff, uint8_t size) {
+#if OSD_ENABLED
+    // use packed decimal characters based on fiam idea implemented in inav osd
+    // search the decimal separator with a bound and always terminate the string
+    char* p = (char*)memchr(&buff[1],'.',size-1);
```

Comment:
This will cause illegal accesses if size is zero.

We should add an explicit check in - just for the no-osd case as that will change a random byte on the stack...

Reply:
you're absolutely right! will do